packages/components/src/languageSupport/squiggle.ts

```diff
@@ -4,86 +4,134 @@ import {
 
-        LambdaSyntax: t.blockComment,
+              LambdaSyntax: t.blockComment,
 
-        VariableName: t.constant(t.variableName),
-        Field: t.variableName,
-        LambdaParameter: t.variableName,
-      }),
-      foldNodeProp.add({
-        LambdaExpr: (context) => ({
-          from: context.getChild("NonEmptyProgram")?.from || 0,
-          to: context.getChild("NonEmptyProgram")?.to || 0,
+              VariableName: t.constant(t.variableName),
+              Field: t.variableName,
+              LambdaParameter: t.variableName,
+            }),
+            foldNodeProp.add({
+              LambdaExpr: (context) => ({
+                from: context.getChild("NonEmptyProgram")?.from || 0,
+                to: context.getChild("NonEmptyProgram")?.to || 0,
+              }),
+              BlockExpr: foldInside,
+              RecordExpr: foldInside,
+              ArrayExpr: foldInside,
+            }),
+            indentNodeProp.add({
+              RecordExpr: (context) =>
+                context.baseIndent +
+                (context.textAfter === "}" ? 0 : context.unit),
+              BlockExpr: (context) =>
+                context.baseIndent +
+                (context.textAfter === "}" ? 0 : context.unit),
+              LambdaExpr: (context) =>
+                context.baseIndent +
+                (context.textAfter === "}" ? 0 : context.unit),
+              ArrayExpr: (context) =>
+                context.baseIndent +
+                (context.textAfter === "]" ? 0 : context.unit),
+            }),
+          ],
         }),
-        BlockExpr: foldInside,
-        RecordExpr: foldInside,
-        ArrayExpr: foldInside,
-      }),
-      indentNodeProp.add({
-        RecordExpr: (context) =>
-          context.baseIndent + (context.textAfter == "}" ? 0 : context.unit),
-        BlockExpr: (context) =>
-          context.baseIndent + (context.textAfter == "}" ? 0 : context.unit),
-        LambdaExpr: (context) =>
-          context.baseIndent + (context.textAfter == "}" ? 0 : context.unit),
-        ArrayExpr: (context) =>
-          context.baseIndent + (context.textAfter == "]" ? 0 : context.unit),
-      }),
-    ],
-  }),
-  languageData: {
-    commentTokens: {
-      line: "#",
-    },
-    autocomplete: [
-      snippetCompletion("{ |${args}| ${body} ", {
-        label: "{",
-        detail: "lambda function",
-      }),
-    ],
-    closeBrackets: {
-      brackets: ['"', "'", "(", "{"],
-    },
-  },
-});
+        languageData: {
+          commentTokens: {
+            line: "#",
+          },
+          autocomplete: (cmpl: CompletionContext) => {
+            const tree = syntaxTree(cmpl.state);
+            const tag = tree.cursorAt(cmpl.pos);
+            const defaultCompletion = {
+              from: tag.from,
+              options: [
+                /*eslint no-template-curly-in-string: "off"*/
+                snippetCompletion("{ |${args}| ${body} ", {
+                  label: "{",
```

Comment:
(I thought I filed this as a separate issue, but I can't find it.)

Can you make this less aggressive? E.g. change to `{|`. I often type `foo = {<enter>` (for format it in multi-line style) and this snippet fires when I didn't intend it, it's pretty annoying.